### PR TITLE
Fix(aks) Ensure AKS RKE2 default loaded region aligns with Credentials

### PR DESCRIPTION
**Pull request by smallteeths**
https://github.com/rancher/dashboard/issues/10470

### CODE REVIEW

**Comment by richard-cox** [comment]
Tracking Info
https://github.com/rancher/rancher/issues/43608 is the core issue
https://github.com/rancher/rancher/pull/43605 contains the fix backend side
https://github.com/rancher/dashboard/pull/10071 contains fix frontend side

**Comment by mantis-toboggan-md** [comment]
@smallteeths can this be tested before https://github.com/rancher/rancher/pull/43605 is merged?

File: shell/machine-config/azure.vue [hunk 1]
```diff
@@ -509,7 +531,7 @@ export default {
           :searchable="false"
           :required="true"
           :label="t('cluster.machineConfig.azure.environment.label')"
-          :disabled="disabled"
+          :disabled="true"
```
Comment (mantis-toboggan-md) [inline shell/machine-config/azure.vue:534]: If this input is _never_ editable it can be removed.

**Comment by mantis-toboggan-md** [comment]
@smallteeths a very similar-looking issue was reported with azure gov credentials; does the fix for that also address your backend issue? https://github.com/rancher/rancher/pull/43009/files

File: shell/machine-config/azure.vue [hunk 2]
```diff
@@ -389,7 +411,7 @@ export default {
   },
 
   created() {
-    if (this.mode === 'create') {
+    if (this.mode === 'create' || !this.value.nsg) {
```
[the remark this answers is not in the document]
Comment (smallteeths) [reply shell/machine-config/azure.vue]: done, I did indeed overthink things

File: shell/machine-config/azure.vue [hunk 3]
```diff
@@ -92,7 +94,7 @@ const storageTypes = [
     value: 'StandardSSD_LRS'
   }
 ];
-const DEFAULT_REGION = 'westus';
+// const DEFAULT_REGION = 'westus';
```
[the remark this answers is not in the document]
Comment (smallteeths) [reply shell/machine-config/azure.vue]: done

File: shell/machine-config/azure.vue (hunk 1 above)
Comment (smallteeths) [reply shell/machine-config/azure.vue:533]: Deleting here is not appropriate. It needs to be shown to the user to let them know which environment I'm using.

**Comment by smallteeths** [comment]
> @smallteeths a very similar-looking issue was reported with azure gov credentials; does the fix for that also address your backend issue? https://github.com/rancher/rancher/pull/43009/files

Yes, it seems I can close the backend PR.

File: shell/machine-config/azure.vue [hunk 4]
```diff
@@ -156,24 +163,32 @@ export default {
...
+      // // the location field doesn't come populated which causes the vmSizes request
+      // // to return 200 but with a null response (also a bunch of other fields are undefined...)
+      // // so let's prefill them with the defaults
+      // } else if (this.mode === _EDIT && !this.value?.location) {
+      //   for (const key in this.defaultConfig) {
+      //     if (this.value[key] === undefined) {
+      //       this.$set(this.value, key, this.defaultConfig[key]);
+      //     }
+      //   }
+      // }
```
Comment (mantis-toboggan-md) [inline shell/machine-config/azure.vue:188]: I think we still need this block: when adding a machine pool on edit defaults are not set and incorrect errors appear:
Screen Shot 2023-12-14 at 11 57 09 AM

From my testing, the location is still set correctly with this code back in place.

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]

**Comment by mantis-toboggan-md** [comment]
This PR also fixes https://github.com/rancher/dashboard/issues/9858

**Comment by mantis-toboggan-md** [comment]
Closing in favor of https://github.com/rancher/dashboard/pull/10471

**Pull request closed without merging**